stl/stl.py

```diff
@@ -249,10 +249,16 @@ def save(self, filename, fh=None, mode=AUTOMATIC, update_normals=True):
             self.update_normals()
 
         if mode is AUTOMATIC:
-            if fh and os.isatty(fh.fileno()):  # pragma: no cover
-                write = self._write_ascii
-            else:
-                write = self._write_binary
+            # Try to determine if the file is a TTY. If it fails then it's
+            # a io.BytesIO() (or something similar) - default to BINARY in this
+            # case.
+            write = self._write_binary
+            if fh:  # pragma: no cover
+                try:
+                    if os.isatty(fh.fileno()):
+                        write = self._write_ascii
+                except IOError:
+                    pass
```

Comment:
Won't this silently ignore a failing write?

Reply:
It will silently ignore that `os.isatty(fh.fileno())` failed. But `write` will still be set because of the
```python
             write = self._write_binary 
```
line above it. No writing actually happens until further down either way. 

I think I'll restructure that. It took me a bit of staring to work out that I hadn't forgotten to set `write` for either `isatty()` returns false or it fails.